pkg/collector/osmlogs_collector.go

```diff
@@ -41,96 +40,96 @@ func (collector *OsmLogsCollector) Collect() error {
+func callNamespaceCollectors(collector *OsmLogsCollector, monitoredNamespaces, controllerNamespaces []string, rootPath, meshName string) error {
 	for _, namespace := range monitoredNamespaces {
-		if err := collectEnvoyData(collector, rootPath, meshName, namespace); err != nil {
+		namespaceRootPath := filepath.Join(rootPath, "namespace_"+namespace)
+		if err := collectDataFromEnvoys(collector, namespaceRootPath, namespace); err != nil {
 			log.Printf("Failed to collect Envoy configs in OSM monitored namespace %s: %+v", namespace, err)
 		}
+		if err := collectNamespaceResources(collector, namespaceRootPath, namespace); err != nil {
+			log.Printf("Failed to collect resource data for OSM monitored namespace %s: %+v", namespace, err)
+		}
 	}
-
-	for _, namespace := range append(monitoredNamespaces, controllerNamespaces...) {
-		var namespaceResourcesMap = map[string][]string{
-			meshName + "_" + namespace + "_metadata":                 []string{"get", "namespaces", namespace, "-o", "jsonpath={..metadata}", "-o", "json"},
-			meshName + "_" + namespace + "_services_table":           []string{"get", "services", "-n", namespace},
-			meshName + "_" + namespace + "_services_configs":         []string{"get", "services", "-n", namespace, "-o", "json"},
-			meshName + "_" + namespace + "_endpoints_table":          []string{"get", "endpoints", "-n", namespace},
-			meshName + "_" + namespace + "_endpoints_configs":        []string{"get", "endpoints", "-n", namespace, "-o", "json"},
-			meshName + "_" + namespace + "_configmaps_table":         []string{"get", "configmaps", "-n", namespace},
-			meshName + "_" + namespace + "_configmaps_configs":       []string{"get", "configmaps", "-n", namespace, "-o", "json"},
-			meshName + "_" + namespace + "_ingresses_table":          []string{"get", "ingresses", "-n", namespace},
-			meshName + "_" + namespace + "_ingresses_configs":        []string{"get", "ingresses", "-n", namespace, "-o", "json"},
-			meshName + "_" + namespace + "_service_accounts_table":   []string{"get", "serviceaccounts", "-n", namespace},
-			meshName + "_" + namespace + "_service_accounts_configs": []string{"get", "serviceaccounts", "-n", namespace, "-o", "json"},
-			meshName + "_" + namespace + "_pods_list":                []string{"get", "pods", "-n", namespace},
+	for _, namespace := range controllerNamespaces {
+		namespaceRootPath := filepath.Join(rootPath, "controller_namespace_"+namespace)
+		if err := collectPodLogs(collector, namespaceRootPath, namespace); err != nil {
+			log.Printf("Failed to collect pod logs for controller namespace %s: %+v", namespace, err)
 		}
-
-		if err := collectPodConfigs(collector, rootPath, meshName, namespace); err != nil {
-			log.Printf("Failed to collect pod logs for ns %s", namespace, err)
+		if err := collectNamespaceResources(collector, namespaceRootPath, namespace); err != nil {
+			log.Printf("Failed to collect resource data for OSM monitored namespace %s: %+v", namespace, err)
```

Comment:
Can we do `log.Print("Failed to collect resource data)` since `collectNamespaceResources()` will capture the specific log error in L 90/L109?

Reply:
Removing this per our conversation offline because no error is ever returned here